packages/node/src/legacy.api.client.ts

```diff
@@ -0,0 +1,98 @@
+import * as https from 'https';
+import { CloudDirectorAuthentication } from './auth';
+import { VCloudExtensibleType } from '@vcd/bindings/vcloud/api/rest/schema_v1_5';
+
+
+export class LegacyApiClient {
+
+    private readonly doubleSlash = '//';
+    private readonly basePath: string;
+
+    constructor(basePath: string,
+                private cloudDirectorAuthentication: CloudDirectorAuthentication,
+                private rejectUnauthorized: boolean) {
+        // remove the http protocol
+        let path: string;
+        if (basePath.startsWith('http')) {
```

Comment:
here you can use the [URL](https://nodejs.org/api/url.html) to parse the url instead of manually parsing the string.